src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -3639,6 +3649,810 @@ .maxstack  3
+                //     public C(C c) // 1, 2
+                Diagnostic(ErrorCode.ERR_CopyConstructorMustInvokeBaseCopyConstructor, "C").WithArguments("B.B(B)").WithLocation(6, 12)
+                );
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor_DerivesFromObject()
+        {
+            var source =
+@"public record C(int I)
+{
+    public C(C c)
+    {
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            var verifier = CompileAndVerify(comp, verify: ExecutionConditionUtil.IsCoreClr ? Verification.Skipped : Verification.Fails);
+            verifier.VerifyIL("C..ctor(C)", @"
+{
+  // Code size       14 (0xe)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  stfld      ""int C.<I>k__BackingField""
+  IL_0007:  ldarg.0
+  IL_0008:  call       ""object..ctor()""
+  IL_000d:  ret
+}
+");
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor_DerivesFromObject_UsesThis()
+        {
+            var source =
+@"public record C(int I)
+{
+    public C(C c) : this(c.I)
+    {
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            var verifier = CompileAndVerify(comp, verify: ExecutionConditionUtil.IsCoreClr ? Verification.Skipped : Verification.Fails);
+            verifier.VerifyIL("C..ctor(C)", @"
+{
+  // Code size       13 (0xd)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  callvirt   ""int C.I.get""
+  IL_0007:  call       ""C..ctor(int)""
+  IL_000c:  ret
+}
+");
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor_DerivesFromObject_UsesBase()
+        {
+            var source =
+@"public record C(int I)
+{
+    public C(C c) : base()
+    {
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            var verifier = CompileAndVerify(comp, verify: ExecutionConditionUtil.IsCoreClr ? Verification.Skipped : Verification.Fails);
+            verifier.VerifyIL("C..ctor(C)", @"
+{
+  // Code size       14 (0xe)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  stfld      ""int C.<I>k__BackingField""
+  IL_0007:  ldarg.0
+  IL_0008:  call       ""object..ctor()""
+  IL_000d:  ret
+}
+");
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor_NoPositionalMembers()
+        {
+            var source =
+@"public record B(object N1, object N2)
+{
+}
+public record C(object P1) : B(0, 1)
+{
+    public C(C c) // 1, 2
```

Comment:
>public C(C c) // 1, 2 [](start = 4, length = 21)

It doesn't look like there are any tests covering and verifying IL for successfully declared copy constructor in a derived record with field initializers and primary constructor. I suspect initializers are going to be emitted for this scenario too.

Reply:
Primary ctor and explicit field initializers should at least be tested separately because presence of a primary constructor might hide some issues around explicit initializers.

---
In reply to: [439031870](https://github.com/dotnet/roslyn/pull/45022#discussion_r439031870) [](ancestors = 439031870)

Reply:
Adding `CopyCtor_UserDefined_WithFieldInitializers`

---
In reply to: [439031870](https://github.com/dotnet/roslyn/pull/45022#discussion_r439031870) [](ancestors = 439031870)